### ISSUE TITLE
[Bug]: Infinite money mode enabled for old games

### ISSUE THREAD

**Issue by PeterN** [OpenTTD/OpenTTD#12147]
### Version of OpenTTD

14.0rc1

### Expected result

Infinite money mode should be disabled when loading old games before infinite money mode was implemented.

### Actual result

Infinite money mode is enabled when loading old games, if infinite money mode is enabled in new game settings.

### Steps to reproduce

1. Enable infinite money mode in settings.
2. Load an older save game.
3. Infinite money mode is active.

**Comment by TrueBrain** [comment]
If my memory serves me well, this is kinda a funny bug :)

Back when writing table-headers into savegames, we knew that one of the usages would be that you could freely add / remove / move settings, without actually having to bump the savegame or doing difficult, etc. But there was a bit of a discussion, and there were two camps.

Camp A said: don't implement the code yet for adding a new setting, as that is unused code, which might be wrong, as we cannot test it.
Camp B said: let's do it now already, as otherwise we might forget, which can result in the above described bug.

Well, here we are :D

Basically, in `LoadSettings` we read the settings from the savegame, but we never set the settings to their default values, so ones not defined in the savegame go to their default, instead of the last used. It is a relative simple change, by just first iterating over all settings that are about to be loaded, and set them to default. Then load the savegame. That should resolve this issue, and any future issue related to settings being added / removed / moved :)

**Comment by frosch123** [comment]
Is the *default* value always the same as the *old behavior* value?
`afterload.cpp` assigns some *old behavior* values to new settings (`map_height_limit`, `timekeeping_units`, `modified_catchment`, ... plenty more). Maybe it's worthwhile to extend the `.ini` files with a `old_default` item?

**Comment by TrueBrain** [comment]
`afterload.cpp` kicks in after loading of the settings, so that should not have any impact. The savegame version is still old, so those `old_default`s should still be recovered.

That said, unifying them into a single place wouldn't hurt anyone.
